core/src/main/java/apoc/trigger/Trigger.java

```diff
@@ -109,26 +86,80 @@ public Stream<TriggerInfo> list() {
                 );
     }
 
+    // TODO - change with @SystemOnlyProcedure
+    @SystemProcedure
+    @Procedure(mode = Mode.WRITE)
+    @Description("CALL apoc.trigger.install(databaseName, name, statement, selector, config) | add a trigger kernelTransaction under a name, in the kernelTransaction you can use $createdNodes, $deletedNodes etc., the selector is {phase:'before/after/rollback/afterAsync'} returns previous and new trigger information. Takes in an optional configuration.")
+    public Stream<TriggerInfo> install(@Name("databaseName") String databaseName, @Name("name") String name, @Name("kernelTransaction") String statement, @Name(value = "selector")  Map<String,Object> selector, @Name(value = "config", defaultValue = "{}") Map<String,Object> config) {
+        checkProcedureRouted();
+        Util.validateQuery(ApocConfig.apocConfig().getDatabase(databaseName), statement);
```

Comment:
```java
        Util.validateQuery(ApocConfig.apocConfig().getDatabase(databaseName), statement);
```

Please forgive me if I'm wrong but this type of logic will not work in 5.0.0. Basically what the code is doing is that it is beginning a new transaction against the target database in order to made sure the statement is valid. 

The reason that this will not work in 5.0.0 is because this Java code will be executed on the leader of the system database, and that in 5.0.0 not all machines in the cluster will have the all the databases on them. And so, you have to assume that the leader of the system database doesn't have access to this user database.

It only works in 4.4 because all machines in a cluster contain all databases.

My opinion is that this type of logic should not be necessary. It should be up to our users to make sure their queries work in their dev environments before pushing their queries to production.  We shouldn't be running these types of checks in production which make queries slower. 

If we did however want to have this type of logic in 5.x then we'd probably have to use Fabric to make sure the validation gets routed to the correct database. In 5.x I don't see any other option that to remove entirely this type of logic. I'm not sure if that's possible? We may want to ship a simple version of the code first that can be improved later.

Reply:
It makes sense. I might be wrong but, given that this piece of code should be useful to prevent additions with wrong syntaxes, I think we could just do `Util.validateQuery(db, statement);`, where `db` is defined as `@Context public GraphDatabaseService db;`. I think it can be ok, but I could be wrong. Wdyt?

Reply:
As discussed this is fine in 4.4, but wouldn't work in 5.x